### PR TITLE
Fix : Added imageView to the response in the app.

### PR DESCRIPTION
**Pull request by mayank408**
Fixes issue #762 

Changes: Added image view support to the response coming from the server in the form of link.

Screenshots for the change: 

screenshot_1498237168

### CODE REVIEW

**Comment by mayank408** [comment]
@chiragw15 @amitiwary999 pls review.

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 1]
```diff
@@ -259,6 +264,8 @@ public int getItemViewType(int position) {
         if (item.getId() == -404) return DOTS;
         else if (item.getId() == -405) return NULL_HOLDER;
         else if (item.isDate()) return DATE_VIEW;
+        else if (item.getContent() != null && (item.getContent().endsWith(".jpg")||(item.getContent().endsWith(".png"))))
```
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Isn't there a better way to do this?
There are many URLs which don't end with jpg or png, but still have a link to direct image.

Example : https://lh3.googleusercontent.com/0-BzaWtxoAnsBjQ_wzUcKxyF07XE7v2Kkg1ogPVUdzmQpvaz118uHQEGU6BdtzJuzfo=h556

File: app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java [hunk 2]
```diff
@@ -341,11 +342,15 @@ private void parseSusiResponse(SusiResponse susiResponse, int i) {
...
                             .get(i).getExpression();
+                    Log.d(TAG, "parseSusiResponse: " + answer);
                     List<String> urlList = extractUrls(answer);
-                    Log.d(TAG, urlList.toString());
+                    Log.d(TAG + " answerLink", urlList.toString());
                     isHavingLink = urlList != null;
                     if (urlList.size() == 0) isHavingLink = false;
+
                 } catch (Exception e ) {
+                    Log.d(TAG, "parseSusiResponse: " + e.toString());
```
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java]: Log the exception as error and not debug.

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 3]
```diff
@@ -327,9 +334,68 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
...
+                    .into(imageViewHolder.susiImage, new com.squareup.picasso.Callback() {
+                        @Override
+                        public void onSuccess() {
+                            imageViewHolder.susiImage.setVisibility(View.VISIBLE);
+                            imageViewHolder.timestampTextView.setVisibility(View.VISIBLE);
+                        }
+
+                        @Override
+                        public void onError() {
+                            Log.d("Error","image can't loaded");
```
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Log the errors as errors

File: app/src/main/java/org/fossasia/susi/ai/model/ChatMessage.java [hunk 4]
```diff
@@ -354,4 +354,5 @@ public int getCount() {
     public void setCount(int count) {
         this.count = count;
     }
+
```
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/model/ChatMessage.java]: Shouldn't be here.

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java (hunk 1 above)
Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Actually the thing is the link coming from the server are from pixabay.com and they all end with .jpg

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java (hunk 3 above)
Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: okay

**Comment by mayank408** [comment]
@the-dagger updated pls check now.

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java (hunk 1 above)
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Hm, this still is a hack.
Can we update the response object to contain a key which indicates what kind of data it's bringing?
That's a better way to handle this

Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Ya I have discussed this with @dynamitechetan , he will adding the image action soon in the server. Let it be open till one and I will update it.

**Comment by chiragw15** [comment]
@mayank408 I am closing this as the image action type is still being implemented on server. Keep the code in your branch. When the image action type is implemented, just send a PR.

**Pull request closed without merging**